addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -34,14 +34,17 @@ def write(self, vals):
 
     def _create_future_public_holidays_timesheets(self, employees):
         lines_vals = []
+        today = fields.Datetime.today()
+        global_leaves_wo_calendar = self.env['resource.calendar.leaves'].search([('calendar_id', '=', False), ('date_from', '>=', today)])
         for employee in employees:
             if not employee.active:
                 continue
             # First we look for the global time off that are already planned after today
-            global_leaves = employee.resource_calendar_id.global_leave_ids.filtered(lambda l: l.date_from >= fields.Datetime.today())
-            work_hours_data = global_leaves._work_time_per_day()
+            global_leaves = employee.resource_calendar_id.global_leave_ids.filtered(lambda l: l.date_from >= today) + global_leaves_wo_calendar
+            resource_calendars = global_leaves._get_resource_calendars()
```

Comment:
So, with `_work_time_per_day` now having an optional argument, this line is not needed anymore.